src/Sentry.Unity/ScreenshotEventProcessor.cs

```diff
@@ -4,61 +4,22 @@
 
 namespace Sentry.Unity;
 
-public class ScreenshotEventProcessor : ISentryEventProcessorWithHint
+public class ScreenshotEventProcessor : ISentryEventProcessor
 {
     private readonly SentryUnityOptions _options;
-    private readonly IApplication _application;
-    public ScreenshotEventProcessor(SentryUnityOptions sentryOptions) : this(sentryOptions, null) { }
+    private readonly SentryMonoBehaviour _sentryMonoBehaviour;
 
-    internal ScreenshotEventProcessor(SentryUnityOptions sentryOptions, IApplication? application)
-    {
-        _options = sentryOptions;
-        _application = application ?? ApplicationAdapter.Instance;
-    }
+    public ScreenshotEventProcessor(SentryUnityOptions sentryOptions) : this(sentryOptions, SentryMonoBehaviour.Instance) { }
 
-    public SentryEvent? Process(SentryEvent @event)
+    internal ScreenshotEventProcessor(SentryUnityOptions sentryOptions, SentryMonoBehaviour? sentryMonoBehaviour)
     {
-        return @event;
+        _options = sentryOptions;
+        _sentryMonoBehaviour = sentryMonoBehaviour ?? SentryMonoBehaviour.Instance;
```

Comment:
We have a public ctor that passes `SentryMonoBehaviour.Instance` already, having also `??` seems redundant.

Since this is an internal ctor, how about:
```suggestion
        _sentryMonoBehaviour = sentryMonoBehaviour ?? throw new ArgumentNullException("sentryMonoBehaviour");
```